File: inventory/inventory/report/actual_inventory_qty/actual_inventory_qty.py
# Copyright (c) 2013, Myme and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = [], []
	
	columns = ["Item Code:Link/Item:100","Colour:Link/Colour:100","Group:Data:100","Yard/Meter per Roll:Float:100","UOM:Link/UOM:100","Total Roll:Float:100","Total Yard/Meter:Float:100"]
	
	item_clause = ""
	if filters.get("item"):
		item_clause = """ AND mi.`item_code`="{0}" """.format(filters.get("item"))
	
	colour_clause = ""
	if filters.get("colour"):
		colour_clause = """ AND di.`colour`="{0}" """.format(filters.get("colour"))
	
	group_clause = ""
	if filters.get("group"):
		group_clause = """ AND di.`group` LIKE "%{0}" """.format('.' + filters.get("group"))
	
	negative_clause = ""
	if filters.get("is_negative"):
		negative_clause = """ AND di.`total_roll`<0 """
	
	result = frappe.db.sql("""
		SELECT mi.`item_code`,di.`colour`,di.`group`,di.`yard_atau_meter_per_roll`,di.`inventory_uom`,di.`total_roll`,di.`total_yard_atau_meter` 
		FROM `tabMaster Inventory`mi JOIN `tabData Inventory`di ON di.`parent` = mi.`name` WHERE di.`total_roll`!=0
		{0} {1} {2} {3}
		ORDER BY mi.`item_code`,di.`colour`
		""".format(item_clause,colour_clause,group_clause,negative_clause),as_list=1)
	
	data = []
	cur_item = ""
	for res in result :
		if res[2] :
			cur_res2 = res[2].split(".")
			res[2] = cur_res2[-1]
		data.append(res)
	
	return columns, data

File: inventory/inventory/report/actual_inventory_qty/actual_inventory_qty.js
// Copyright (c) 2016, Myme and contributors
// For license information, please see license.txt

frappe.query_reports["Actual Inventory Qty"] = {
	"filters": [
		{
			"fieldname": "item",
			"label":__("Item"),
			"fieldtype": "Link",
			"options": "Item"
		},
		{
			"fieldname": "colour",
			"label":__("Colour"),
			"fieldtype": "Link",
			"options": "Colour"
		},
		{
			"fieldname": "group",
			"label":__("Group"),
			"fieldtype": "Data",
		},
		{
			"fieldname": "is_negative",
			"label":__("Negative Only"),
			"fieldtype": "Check",
		},
	]
}
